src/solver/modeler/ortoolsImpl/include/antares/solver/modeler/ortoolsImpl/linearProblem.h

```diff
@@ -61,11 +61,12 @@ class OrtoolsLinearProblem final: public Api::ILinearProblem
     bool isMaximization() const override;
 
     OrtoolsMipSolution* solve(bool verboseSolver) override;
+    operations_research::MPSolver* MPSolver();
 
 private:
     OrtoolsMipVariable* addVariable(double lb, double ub, bool integer, const std::string& name);
 
-    std::shared_ptr<operations_research::MPSolver> mpSolver_;
+    operations_research::MPSolver* mpSolver_;
```

Comment:
As you can see, **mpSolver_** is no longer a shared pointer, at least for now.
The reason is that current **Antares** code (where the **ortools** solver are called) is made in such a way that the solver must be shared between optimization and weeks.
So a solver cannot be destroyed too soon, it has to be persistent.
It's planned to handle the life time in the future.